### PR TITLE
Fix #11841

### LINKED ISSUE #11841: Using flow_from_dataframe with predict_generator produces wrong ordering of filenames
flow_from_dataframe is a very useful function allowing flexibility that flow_from_directory lacks, for e.g, regression tasks.
But the problem with this function is it processes the dataframe's (filename) column in sorted way.
for example
filename  y
a              3.2
d              2.2
c               5.2

in principle, the generator should process files in this way:
a,d,c
but the current implementation processes file in alphabtic order
a,c,d

this is not so convenient for predict_generator function

### PR DESCRIPTION
**Pull request by fuzzythecat**
### Summary
This fixes #11841. I will work on a follow-up PR on further documentations in keras-preprocessing.
 
### PR Overview

- [n] This PR requires new unit tests [y/n] (make sure tests are included)
- [n] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [y] This PR is backwards compatible [y/n]
- [n] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

**Comment by gabrieldemarmiesse** [comment]
The travis errors are not due to your modifications. Let me investigate.

**Comment by gabrieldemarmiesse** [comment]
Actually, you have trailing whitespaces, which is one cause of the errors but not the only one.

**Comment by gabrieldemarmiesse** [comment]
It seems there are some major breakage between theano and numpy.  All builds are failing because we run some theano tests in other builds (backend tests).

**Comment by gabrieldemarmiesse** [comment]
See #12037

File: keras/preprocessing/image.py [hunk 1]
```diff
@@ -102,6 +102,7 @@ class DirectoryIterator(image.DirectoryIterator, Iterator):
             `None`: no targets get yielded (only input images are yielded).
         batch_size: Integer, size of a batch.
         shuffle: Boolean, whether to shuffle the data between epochs.
+            If set to False, sorts the data in alphabetical order.
```
Comment (fchollet) [inline keras/preprocessing/image.py]: alphanumeric

**Comment by fuzzythecat** [comment]
@gabrieldemarmiesse @fchollet Made the changes.

**Comment by nouiz** [comment]
We just released Theano 1.0.4 that added support for numpy 1.16 on the CPU. We didn't test on the GPU.

**Pull request merged**